Add embed block placeholder.

BE-Webdesign opened the pull request:
Closes #551.

This PR adds a placeholder for the embed block.  Currently it does not
function and merely serves as a step forward on the UI.

**Testing Instruction:**

Open the inserter menu and add an embed block to the editor.  The embed
block should feature an input to submit the URL.  This functionality
does not yet work but serves as a placeholder.

**Note:**

Someone needs to do the translation build, as Windows breaks it.

BE-Webdesign committed:
Add embed block placeholder.

Closes #551.

This PR adds a placeholder for the embed block.  Currently it does not
function and merely serves as a step forward on the UI.

**Testing Instruction:**

Open the inserter menu and add an embed block to the editor.  The embed
block should feature an input to submit the URL.  This functionality
does not yet work but serves as a placeholder.

**Note:**

Someone needs to do the translation build, as Windows breaks it.

jasmussen commented:
Nice work! ⭐️ 

Would you like some UI for that, and if so would you prefer I do it in a separate branch post-merge?

BE-Webdesign commented:
>Would you like some UI for that

Yes please :smile:

> and if so would you prefer I do it in a separate branch post-merge?

Whatever works best for you.

jasmussen committed:
Polish the visual styles a bit.

- Add a bg color. This will receive further love later, as this block has surfaced a problem with the initial design.
- Use the default font stack
- Normalize the input fields for gutenberg
- Rephrase ever so slightly
- Reposition input field and embed button with flex

jasmussen commented:
I pushed a little polish 👆

![screen shot 2017-05-02 at 18 37 23](https://cloud.githubusercontent.com/assets/1204802/25628498/ad9e6a84-2f66-11e7-8919-95b44cc5b00a.png)

Thank you again!

mtias commented on blocks/library/embed/index.js:
```diff
@@ -22,6 +26,23 @@ registerBlock( 'core/embed', {
 	edit( { attributes, setAttributes, focus, setFocus } ) {
 		const { url, title, caption } = attributes;
 
+		if ( ! url ) {
```
@aduth thoughts on a separate default method for this? I think it would clean up edit implementation a bit.

mtias replied:
I mean more moving the return into a separate function.

BE-Webdesign replied:
Oh yeah, I misunderstood. We should do that at some point for sure.  I don't think we need to right now, but once the placeholder actually has to do something we will probably turn it into its own component.

aduth replied:
I don't know that this necessarily needs to be built-in, but I think we could surface better the fact that `edit` is truly a React component, and can itself branch into separate subcomponents up to the implementer's discretion.

mtias commented on blocks/library/embed/index.js:
```diff
@@ -22,6 +26,23 @@ registerBlock( 'core/embed', {
 
+		if ( ! url ) {
+			return (
+				<div className="blocks-embed is-placeholder">
```
This sounds like a `<Placeholder>` reusable component. @jasmussen do we expect the design to share these commonalities between image/embed/video/etc placeholders?

nylen replied:
Regarding this question and the above about splitting out into a separate method... I'm not sure we can know the answer until we implement the placeholders for a couple of other blocks.  I vote we merge something like this structure and iterate later.

BE-Webdesign replied:
The interface for embedding a URL and inserting from the media library will probably be different.

jasmussen replied:
> This sounds like a <Placeholder> reusable component. @jasmussen do we expect the design to share these commonalities between image/embed/video/etc placeholders?

Yes, the design for a placeholder block is always 90% the same. Always a block, always an icon and a title centered, and always some more options after that. Those more options are likely going to be different, as @BE-Webdesign suggests:

- Image will have a "open media library" button
- Embeds will have a URL option
- We'll probably run into other things as well.

So maybe something like, and please correct my pseudo code:

```
<Placeholder icon="cloud" title="Embed>
... input field and button ...
</Placeholder>
```

BE-Webdesign replied:
This could probably make use of the Slot Fill as well.  But I think that would be a future issue.

nylen committed:
Update translations

BE-Webdesign commented:
@nylen #597.

nylen commented:
@BE-Webdesign beware that I rebased and force-pushed this branch to fix the translation build (again) and the resulting conflicts with `master`.

+1 for merging this soon, but what do you think about the remaining questions in https://github.com/WordPress/gutenberg/pull/603#issuecomment-298694612 (what's left to actually make this work?)

BE-Webdesign commented:
I think making it actually work is best left to another PR.  I can however do it in this one if need be.

We would need to handle the input, validate that a URL was entered as the input, handle a successful switch to the embedded content, and handle any errors in case the user enters an invalid embed URL that leads no where.  I am unfamiliar with using iFrames and React and their performance characteristics, so if it seems choppy, potentially some performance tweaks will be necessary as well.  We will also need a way to change the url on the "final" version, and apply the same input/error handling there.

BE-Webdesign commented:
Merge? Making this fully functional is kind of a big step forward and depends on other moving parts in the project.

nylen commented:
Agreed. Thanks for working on this!

aduth commented on editor/assets/stylesheets/main.scss:
```diff
@@ -48,4 +48,25 @@ body.toplevel_page_gutenberg {
 		padding-top: $header-height;
 	}
+
+	input, textarea {
```
This style is overriding the Text mode textarea styles, eliminating padding and changing font appearance.

jasmussen replied:
Ack, my bad, will fix.
